packages/storage/compat/list.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@firebase/storage-types';
+import { ListResult } from '../src/list';
+import { ReferenceCompat } from './reference';
+import { Reference } from '../src/reference';
+
+export class ListResultCompat implements types.ListResult {
+  constructor(
+    private readonly delegate: ListResult,
+    private converter: (ref: Reference) => ReferenceCompat
```

Comment:
It seems like it doesn't complain about cyclic references if it's only a type, but it does if you bring in actual code. Because types don't impact execution? Not sure the exact rationale.

Reply:
Since this type cannot be constructed without passing in a factory function for References, it seems that in practice, these types are always included in the bundle?